src/classes/CRLP_Rollup_SEL_TEST.cls

```diff
@@ -48,59 +48,748 @@ public class CRLP_Rollup_SEL_TEST {
+            'CRLP_Rollup_SEL.UNSUPPORTED_OPERATIONS_BY_ROLLUP_FIELD has unexpected values'
+        );
+    }
+
+    /**
+     * @description Validate the validateReferencedFields() method to ensure it properly identifies bad fields
+     */
+    @IsTest
+    private static void shouldValidateFieldDescribes() {
+        Map<String, Schema.DescribeFieldResult> referencedFields = new Map<String, Schema.DescribeFieldResult>();
+        Map<String, Schema.DescribeFieldResult> allFields = UTIL_Describe.getAllFieldsDescribe('Account');
+
+        referencedFields.put('name', allFields.get('name'));
+        CRLP_Rollup_SEL.validateReferencedFields(referencedFields, 'Account');
+        System.assert(true, 'An exception was not thrown. Good!');
+
+        try {
+            referencedFields.put('nothing', allFields.get('nothing'));
+            CRLP_Rollup_SEL.validateReferencedFields(referencedFields, 'Account');
+            System.assert(false, 'An exception was not thrown. Bad!');
+
+        } catch (Exception ex) {
+            System.assert(true, 'An exception was not thrown. Good!');
+        }
+    }
+
+    /**
+     * @description Validate the validateReferencedFields() method to ensure it properly identifies bad fields
+     */
+    @IsTest
+    private static void shouldReturnExpectedRollupDefinitions() {
+        mockRollupCMTValues();
+
+        List<Rollup__mdt> rollupsAccountHC = CRLP_Rollup_SEL.getRollupDefinitionsToUse(
+            CRLP_RollupProcessingOptions.RollupType.AccountHardCredit,
+            CRLP_RollupProcessingOptions.RollupTypeFilter.All);
+        System.assertEquals(2, rollupsAccountHC.size(), 'There should be exactly two Rollups returned for AccountHardCredit');
+
+        List<Rollup__mdt> rollupsContactHC = CRLP_Rollup_SEL.getRollupDefinitionsToUse(
+            CRLP_RollupProcessingOptions.RollupType.ContactHardCredit,
+            CRLP_RollupProcessingOptions.RollupTypeFilter.All);
+        System.assertEquals(0, rollupsContactHC.size(), 'There should be exactly zero Rollup returned for ContactHardCredit');
+    }
+
+
+    /**
+     * @description Tests all secenarios for CRLP_Rollup_SEL.getReferencedFieldsByObject(SObjectType objType) 
+     * by storing each scenario in a GetReferencedFieldsByObjectTestData.
+     * Scenario Types:
+     *      Returns empty list: null objType
+     *      Returns special Fields if Opportunity/CurrencyIsoCode: objType doesn't match or no active cachedRollups
+     *      Returns special Fields if Opportunity/CurrencyIsoCode/Cached Rollup: single Rollup Relationship
+     *      Returns special Fields if Opportunity/CurrencyIsoCode/Cached Rollup: all Rollup Relationships and no duplicates
+     */
+    @IsTest
+    private static void testGetReferencedFieldsByObject() {
+        final List<ReferencedFieldsByObjectTestData> allTestData = new List<ReferencedFieldsByObjectTestData>();
+        final String ROLLUP_SOBJECT_NAME = Schema.SObjectType.Rollup__mdt.getName();
```

Comment:
Since this is a local var, we can write it as `rollupSObjectName`.
Similarly, `rollupOperationField`, `rollupActiveField`, `rollupSummaryObjectRelationship`, etc...